pennylane/templates/subroutines/temporary_and.py

```diff
@@ -0,0 +1,253 @@
+
+
+    **Example**
+
+    .. code-block::
+
+        dev = qml.device("default.qubit", shots=1)
+        @qml.qnode(dev)
+        def circuit():
+            qml.X(0)
+            qml.X(1)
+            qml.TemporaryAnd([0,1,2])
+            qml.CNOT([2,3])
+            qml.adjoint(qml.TemporaryAnd([0,1,2])) # We can apply the adjoint TemporaryAnd because after applying a Toffoli,
+                                                   # the target wire would be |0>.
+
+            return qml.sample(wires=[0,1,2,3])
+
+    .. code-block:: pycon
+
+        >>> print(circuit())
+        [1 1 0 1]
+    """
+
+    num_wires = 3
+    """int: Number of wires that the operator acts on."""
+
+    num_params = 0
+    """int: Number of trainable parameters that the operator depends on."""
+
+    ndim_params = ()
+    """tuple[int]: Number of dimensions per trainable parameter that the operator depends on."""
+
+    resource_keys = set(["control_values"])
+
+    def __init__(self, wires: WiresLike, control_values=(1, 1), id=None):
+        wires = Wires(wires)
+        self.hyperparameters["control_values"] = tuple(control_values)
+        super().__init__(wires=wires, id=id)
+
+    @property
+    def resource_params(self) -> dict:
+        return {"control_values": self.hyperparameters["control_values"]}
+
+    def _flatten(self):
+        return tuple(), (self.wires, self.hyperparameters["control_values"])
+
+    @classmethod
+    def _unflatten(cls, _, metadata):
+        return cls(wires=metadata[0], control_values=metadata[1])
+
+    @classmethod
+    def _primitive_bind_call(cls, *args, **kwargs):
+        return cls._primitive.bind(*args, **kwargs)
+
+    @staticmethod
+    @lru_cache()
+    def compute_matrix(**kwargs):  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a canonical matrix in the computational basis (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        Returns:
+            array_like: matrix
+
+        **Example**
+
+        >>> print(qml.TemporaryAnd.compute_matrix())
+        [[ 1.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j]
+         [ 0.+0.j -0.-1.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j]
+         [ 0.+0.j  0.+0.j  1.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j]
+         [ 0.+0.j  0.+0.j  0.+0.j -0.-1.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j]
+         [ 0.+0.j  0.+0.j  0.+0.j  0.+0.j  1.+0.j  0.+0.j  0.+0.j  0.+0.j]
+         [ 0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+1.j  0.+0.j  0.+0.j]
+         [ 0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j -0.-1.j]
+         [ 0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j  1.+0.j  0.+0.j]]
+        """
+
+        control_values = kwargs["control_values"]
+        eye = np.eye(2)
```

Comment:
Couldn't see import of numpy for doing this.

Reply:
I was just changing that part based on a David comment
Right now it is working. I added test to check the 4 combinations